CIP-dapp-bridge-rework/README.md

```diff
@@ -0,0 +1,1169 @@
+        * [getDelegation()](#getdelegationaccountid-accountid-promisedelegation)
+        * [buildTransaction()](#buildtransactiontx-transactionscript-accountid-accountid-promise-transaction-cbortransaction-txhash-hexstring-)
+        * [signTransaction()](#signtransactiontxhash-hexstring-accountid-accountid-outerwitnesses-cborwitnessset-promisecborwitnessset)
+        * [submitTransaction()](#submittransactiontxhash-hexstring-accountid-accountid-promisehexstring)
+        * [isTransactionAccepted()](#istransactionacceptedtxhash-hexstring-promisenumber)
+        * [signData()](#signdatapayload-hexstring-accountid-accountid-withstakekey-boolean-promisedatasignature)
+    * [Example](#example)
+  * [Rationale: how does this CIP achieve its goals?](#rationale-how-does-this-cip-achieve-its-goals)
+  * [Path to Active](#path-to-active)
+    * [Acceptance Criteria](#acceptance-criteria)
+    * [Implementation Plan](#implementation-plan)
+  * [Copyright](#copyright)
+<!-- TOC -->
+
+## Abstract
+
+This document describes the reevaluation of the existing approach to connecting Cardano DApps and wallets, as detailed in CIP-30. Its purpose is to simplify the integration of ecosystem products for both sides, and also to ease the development of DApps themselves by simplifying the process without losing any capabilities.
+The new approach still relies on code injection, but offers a more transparent API. The main idea is that a DApp should not perform the functions of a wallet.
+
+### What is it?
+
+It's automation, lowering the barrier to entry, separation of responsibilities, and scaling the complexity of development.
+
+### What is not?
+
+It's not "getting rid of CBOR", not transferring all functions to the wallet, not replacing everything with JSON.
+
+## Motivation: why is this CIP necessary?
+
+This CIP addresses several issues that complicate the development of the Cardano ecosystem with the current integration standard:
+
+1. **Complexity in Writing Client-Side Logic for DApps**: As they essentially need to implement their own wallet each time, including algorithms for coin selection, UTxO management, UTxO validation, TX building, etc. They shouldn't be doing this. There is already a separately written wallet that connects to their application for this purpose. <br><br>
+
+2. **Integration Difficulties**: This implies complicating the internal logic of wallets, adding endpoints for importing transactions, writing logic for checks/validation of foreign transactions, etc. This could be avoided if the transaction was created and processed natively by the wallet.<br><br>
+
+3. **Privacy Issues**: Currently, the wallet must pass all information to the DApp, including the UTxO set and full addresses on one hand, and trust the user's funds to the transaction building code inside the DApp on the other. This is not obvious to the user, and in case of problems, they will blame the wallet developers.<br><br>
```

Comment:
This is not a privacy problem, because there is no way to issue a transaction without revealing your address anyway

Reply:
Wallets do not reveal addresses from other accounts

Reply:
No, it's a privacy problem, because the 3rd party software gets all my data. They have my addresses (even for an enterprise account with "not bunchable" addresses because of lack of a stake part). They have my UTxO set. Doesn't matter if I use a new address for every new transaction, they have it all. They can track everything they want then. And I can't control it with the CIP30. I can't prevent it. And I can't hide it then, but they can share it to everyone. So it's almost the same as if I had posted my public keys on Twitter.